package/src/components/Chat/hooks/handleEventToSyncDB.ts

```diff
@@ -120,15 +120,35 @@ export const handleEventToSyncDB = async <
   }
 
   if (type === 'message.new') {
-    const message = event.message;
+    const { cid, message, user } = event;
 
     if (message && (!message.parent_id || message.show_in_channel)) {
-      return await queriesWithChannelGuard((flushOverride) =>
-        upsertMessages({
+      return await queriesWithChannelGuard(async (flushOverride) => {
+        let queries = await upsertMessages({
           flush: flushOverride,
           messages: [message],
-        }),
-      );
+        });
+        if (cid && client.user && client.user.id !== user?.id) {
+          const userId = client.user.id;
+          const channel = client.activeChannels[cid];
+          const ownReads = channel?.state.read[userId];
+          const unreadCount = channel?.state.unreadCount;
```

Comment:
We have to use `channel.state.unreadCount` here rather instead of `ownReads.unread_count`, as the `state` one takes mutes, message shadowing and a bunch of other rules into account. The other one does not.